### PR TITLE
windows: ignore spurious mouse move messages

### PR DESCRIPTION
**Pull request by filnet**
- [x] Tested on all platforms changed
- [x] Compilation warnings were addressed
- [x] `cargo fmt` has been run on this branch
- [ ] `cargo doc` builds successfully
- [ ] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [ ] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [ ] Created or updated an example program if it would help users understand this functionality
- [ ] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

### CODE REVIEW

File: src/platform_impl/windows/window_state.rs [hunk 1]
```diff
@@ -48,6 +48,7 @@ pub struct MouseProperties {
     pub cursor: CursorIcon,
     pub buttons_down: u32,
     cursor_flags: CursorFlags,
+    pub last_position: Option<(f64, f64)>,
```
Comment (goddessfreya) [inline src/platform_impl/windows/window_state.rs]: Should be `PhysicalPosition` (or `LogicalPosition`, but I think most likely the former).

File: src/platform_impl/windows/event_loop.rs [hunk 2]
```diff
@@ -857,16 +857,29 @@ unsafe extern "system" fn public_window_callback<T: 'static>(
...
-                    modifiers: event::get_key_mods(),
-                },
-            });
+            let mouse_moved;
+            {
+                // handle spurious WM_MOUSEMOVE messages
+                // see https://devblogs.microsoft.com/oldnewthing/20031001-00/?p=42343
+                // and http://debugandconquer.blogspot.com/2015/08/the-cause-of-spurious-mouse-move.html
+                let mut w = subclass_input.window_state.lock();
+                mouse_moved = match w.mouse.last_position {
```
Comment (goddessfreya) [inline src/platform_impl/windows/event_loop.rs]: `if w.mouse.last_position != Some(position)`? Why a match?

File: src/platform_impl/windows/window_state.rs (hunk 1 above)
Comment (filnet) [reply src/platform_impl/windows/window_state.rs]: You mean Option<PhysicalPosition<P>> ?
I considered it but wouldn't that cause the P to trickle up MouseProperties, WindowState, etc ?

Comment (filnet) [reply src/platform_impl/windows/window_state.rs]: I went for `Option<PhysicalPosition<f64>>`

File: src/platform_impl/windows/event_loop.rs (hunk 2 above)
Comment (filnet) [reply src/platform_impl/windows/event_loop.rs]: done

**Comment by filnet** [comment]
> Needs a changelog entry.

Done

**Pull request merged**